Branch-SDK/src/androidTest/java/io/branch/referral/ReferringUrlUtilityTests.kt

```diff
@@ -178,6 +180,82 @@ class ReferringUrlUtilityTests : BranchTest() {
         assertTrue(areJSONObjectsEqual(expected, params))
     }
 
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.getDefault())
+    @Test
+    fun testSerializeToJson() {
```

Comment:
A future improvement would be to properly serialize/deserialize when we write to a real database (not the sharedpreferences, but datastore)

https://kotlinlang.org/docs/serialization.html